hedera-node/hedera-app/src/main/java/com/hedera/node/app/bbm/contracts/ContractBytecodesDumpUtils.java

```diff
@@ -0,0 +1,273 @@
+import com.hedera.hapi.node.state.contract.Bytecode;
+import com.hedera.hapi.node.state.token.Account;
+import com.hedera.node.app.bbm.DumpCheckpoint;
+import com.hedera.node.app.bbm.accounts.AccountDumpUtils;
+import com.hedera.node.app.bbm.accounts.HederaAccount;
+import com.hedera.node.app.bbm.utils.Writer;
+import com.hedera.node.app.service.contract.ContractService;
+import com.hedera.node.app.service.contract.impl.state.InitialModServiceContractSchema;
+import com.hedera.node.app.service.mono.state.adapters.VirtualMapLike;
+import com.hedera.node.app.service.mono.state.migration.AccountStorageAdapter;
+import com.hedera.node.app.service.mono.state.virtual.EntityNumVirtualKey;
+import com.hedera.node.app.service.mono.state.virtual.VirtualBlobKey;
+import com.hedera.node.app.service.mono.state.virtual.VirtualBlobValue;
+import com.hedera.node.app.service.mono.state.virtual.entities.OnDiskAccount;
+import com.hedera.node.app.service.mono.utils.NonAtomicReference;
+import com.hedera.node.app.spi.state.ReadableKVState;
+import com.hedera.node.app.spi.state.StateDefinition;
+import com.hedera.node.app.state.merkle.StateMetadata;
+import com.hedera.node.app.state.merkle.disk.OnDiskKey;
+import com.hedera.node.app.state.merkle.disk.OnDiskValue;
+import com.hedera.node.app.state.merkle.memory.InMemoryKey;
+import com.hedera.node.app.state.merkle.memory.InMemoryValue;
+import com.hedera.node.app.state.merkle.memory.InMemoryWritableKVState;
+import com.swirlds.merkle.map.MerkleMap;
+import com.swirlds.virtualmap.VirtualMap;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HexFormat;
+import java.util.List;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+
+public class ContractBytecodesDumpUtils {
+
+    private static final SemanticVersion CURRENT_VERSION = new SemanticVersion(0, 47, 0, "SNAPSHOT", "");
+
+    private ContractBytecodesDumpUtils() {
+        // Utility class
+    }
+
+    public static void dumpModContractBytecodes(
+            @NonNull final Path path,
+            @NonNull final VirtualMap<OnDiskKey<AccountID>, OnDiskValue<Account>> accounts,
+            @NonNull final DumpCheckpoint checkpoint) {
+        final var dumpableAccounts = AccountDumpUtils.gatherAccounts(accounts, HederaAccount::fromMod);
+        final var contracts = getModContracts(dumpableAccounts);
+        final var sb = generateReport(contracts);
+        try (@NonNull final var writer = new Writer(path)) {
+            writer.writeln(sb.toString());
+            System.out.printf(
+                    "=== mono contract bytecodes report is %d bytes at checkpoint %s%n",
+                    writer.getSize(), checkpoint.name());
+        }
+    }
+
+    private static Contracts getModContracts(HederaAccount[] dumpableAccounts) {
```

Comment:
Consider moving to `ContractUtils.java` - renaming the current `getContracts` to `getMonoContracts`.

Also would be nice - but not required by any means! - if all or most of `ContractUtils.getAllContractContents` could be reused or adapted.

Reply:
Okay, I've moved the method inside `ContractUtils` 👍 
I didn't get the second part. What do you mean by reusing or adapting the `ContractUtils.getAllContractContents`? Are we using the same method somewhere else?

Reply:
looks like a bunch of very similar code ... probably _not_ worth changing at this stage.